src/org/exist/xquery/lib/kwic.xql

```diff
@@ -156,6 +157,40 @@ function kwic:string-length($nodes as item()*) as xs:integer {
 		0
 };
 
+declare
+    %private
+function kwic:collapse-whitespace-fn($callback as (function(text(), xs:string) as text()?)?) as (function(text(), xs:string) as text()) {
+    function($text as text(), $mode as xs:string) as text()? {
+        let $text :=
+            if(matches($text, "\s+")) then
+                text { " " }
+            else
+                $text
+        return
+            if(exists($callback))then
+                $callback($text, $mode)
+            else
+                $text
+    }
+};
+
+declare
+    %private
+function kwic:drop-whitespace-fn($callback as (function(text(), xs:string) as text()?)?) as (function(text(), xs:string) as text()?) {
+    function($text as text(), $mode as xs:string) as text()? {
+        let $text :=
+            if(matches($text, "\s+")) then
+                ()
+            else
+                $text
+        return
+            if(exists($text) and exists($callback))then
```

Comment:
Whitespace is preferred around the condition of the `if` clause. Not that this is a bible, but http://xqdoc.org/xquery-style.pdf section 6.1 is good guidance.